### PR TITLE
Add useful error msg in `GodotSoftBody3D::set_mesh`

### LINKED ISSUE #91690: Soft body does nothing if it's Mesh is array mesh programmatically created with  SurfaceTool
### Tested versions

A soft body with a procedurally generated mesh does not do anything. When you run the game the mesh just appears at coordinates 0,0,0,

image

### System information

Windows 10 - Godot 4.2.2 stable

### Issue description

E 0:00:00:0471   create_from_trimesh: Condition "p_indices.is_empty()" is true. Returning: false
 <C++ Source>   servers/physics_3d/godot_soft_body_3d.cpp:469 @ create_from_trimesh()

### Steps to reproduce

-Just run the scene and you will see that the volleyball net is static at 0,0,0 coordinates, that's it. Also there is an error in the console: 

### Minimal reproduction project (MRP)

[ProceduralSoftBodyBug.zip](https://github.com/godotengine/godot/files/15242612/ProceduralSoftBodyBug.zip)

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/91690

### CODE REVIEW

File: servers/physics_3d/godot_soft_body_3d.cpp [hunk 1]
```diff
@@ -595,7 +600,7 @@ void GodotSoftBody3D::generate_bending_constraints(int p_distance) {
 		const uint32_t adj_size = n * n;
 		unsigned *adj = memnew_arr(unsigned, adj_size);
 
-#define IDX(_x_, _y_) ((_y_) * n + (_x_))
+#define IDX(_x_, _y_) ((_y_)*n + (_x_))
```
Comment (akien-mga) [inline servers/physics_3d/godot_soft_body_3d.cpp:603]: You might have an older clang-format version than the one used on CI / by `pre-commit`. So this change needs to be reverted to pass CI.

Comment (jsjtxietian) [reply servers/physics_3d/godot_soft_body_3d.cpp:603]: Ah yes, I was about to update it. Thanks!

**Review by akien-mga — APPROVED** [review]

File: servers/physics_3d/godot_soft_body_3d.cpp [hunk 2]
```diff
@@ -140,7 +140,12 @@ void GodotSoftBody3D::set_mesh(RID p_mesh) {
 	Array arrays = RenderingServer::get_singleton()->mesh_surface_get_arrays(soft_mesh, 0);
 	ERR_FAIL_COND(arrays.is_empty());
 
-	bool success = create_from_trimesh(arrays[RenderingServer::ARRAY_INDEX], arrays[RenderingServer::ARRAY_VERTEX]);
+	const Vector<int> &p_indices = arrays[RenderingServer::ARRAY_INDEX];
```
[the remark this answers is not in the document]
Comment (jsjtxietian) [reply servers/physics_3d/godot_soft_body_3d.cpp:143]: Thanks!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**